homeassistant/components/habitica/__init__.py

```diff
@@ -141,19 +142,20 @@ async def handle_api_call(call: ServiceCall) -> None:
 
     data = hass.data.setdefault(DOMAIN, {})
     config = entry.data
-    websession = async_get_clientsession(hass)
+    websession = async_get_clientsession(
+        hass, verify_ssl=config.get(CONF_VERIFY_SSL, True)
+    )
     url = config[CONF_URL]
     username = config[CONF_API_USER]
     password = config[CONF_API_KEY]
     name = config.get(CONF_NAME)
     config_dict = {"url": url, "login": username, "password": password}
     api = HAHabitipyAsync(config_dict)
     user = await api.user.get()
-    if name is None:
-        name = user["profile"]["name"]
+    if name != user["auth"]["local"]["username"]:
         hass.config_entries.async_update_entry(
             entry,
-            data={**entry.data, CONF_NAME: name},
+            data={**entry.data, CONF_NAME: user["auth"]["local"]["username"]},
         )
```

Comment:
Honestly, we should get rid of saving the name in the entry data

Reply:
yeah, the service somehow still needs it, otherwise it is not used anywhere

Reply:
i got rid of the username 🎉

Reply:
not possible until we got rid of the api service, so leaving it for now